hail/python/hailtop/batch/job.py

```diff
@@ -1107,15 +1070,31 @@ def wrapped(*args, **kwargs):
 import traceback
 import json
 import sys
-
-with open(\\"{result}\\", \\"wb\\") as dill_out:
+from time import perf_counter
+
+def deserialize_argument(arg):
+    typ, val = arg
+    if typ == 'py_path':
+        return dill.load(open(val, 'rb'))
+    if typ in ('path', 'dict_path'):
+        return val
+    assert typ == 'value'
+    return val
+
+with open('{result}', 'wb') as dill_out:
     try:
-        with open(\\"{code}\\", \\"rb\\") as f:
-            result = dill.load(f)()
-            dill.dump(result, dill_out, recurse=True)
-            {json_write}
-            {str_write}
-            {repr_write}
+        func = args = kwargs = None
```

Comment:
These don't need to be set outside the context-manager

Reply:
Removed!